src/frames.ts

```diff
@@ -118,22 +117,21 @@ export class FrameManager {
     frame._url = url;
     frame._name = name;
     frame._lastDocumentId = documentId;
+    frame._documentWatchers.forEach(watcher => watcher(documentId));
     this.clearFrameLifecycle(frame);
     this.clearWebSockets(frame);
-    if (!initial) {
-      for (const watcher of this._lifecycleWatchers)
-        watcher._onCommittedNewDocumentNavigation(frame);
+    // TODO should the document watchers only fire if !initial?
```

Comment:
This was important for webkit, otherwise we used to resolve waitForNavigation() with about:blank on cross-process navigation.

Reply:
That doesn't appear to be the case. I added a test.